frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalarSettingsWidgets.tsx

```diff
@@ -0,0 +1,225 @@
+  type: string;
+}
+
+export function PeriodsAgoInputWidget({
+  isSelected,
+  maxValue,
+  name,
+  onChange,
+  selectedValue,
+  setOpen,
+  type,
+}: PeriodsAgoInputWidget) {
+  const value = useMemo(() => {
+    if (!selectedValue) {
+      return null;
+    }
+
+    if (Number.isInteger(selectedValue.value)) {
+      return selectedValue.value;
+    }
+
+    return null;
+  }, [selectedValue]);
+  const minValue = 2;
+
+  const [inputValue, setInputValue] = useState(value ?? minValue);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  // used to prevent accidental button click when mouseDown inside input field
+  // but dragged to highlight all text and accidentally mouseUp on the button
+  // outside the input field
+  const mouseDownInChildRef = useRef(false);
+  const handleChildMouseDownAndUp = useCallback(
+    (e: MouseEvent<HTMLInputElement>) => {
+      e.stopPropagation();
+      e.preventDefault();
+      inputRef.current?.select();
+
+      mouseDownInChildRef.current = true;
+    },
+    [],
+  );
+  const handleParentMouseDown = useCallback(() => {
+    mouseDownInChildRef.current = false;
+  }, []);
+
+  const submitValue = useCallback(() => {
+    if (inputValue < minValue) {
+      return setInputValue(minValue);
+    }
+
+    if (inputValue > maxValue) {
+      return setInputValue(maxValue);
+    }
+
+    if (!Number.isInteger(inputValue)) {
+      return setInputValue(value ?? minValue);
+    }
+
+    onChange({
+      type,
+      value: inputValue,
+    });
+
+    setOpen(false);
+  }, [inputValue, maxValue, type, onChange, setOpen, value]);
+
+  const handleButtonClick = useCallback(
+    (e: MouseEvent<HTMLDivElement>) => {
+      e.stopPropagation();
+      e.preventDefault();
+
+      if (mouseDownInChildRef.current) {
+        mouseDownInChildRef.current = false;
+        return;
+      }
+
+      submitValue();
+    },
+    [submitValue],
+  );
+
+  const handleEnter = (e: KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      mouseDownInChildRef.current = false;
+      inputRef.current?.blur();
+
+      submitValue();
+
+      // re-select input just in case the input number was not a valid input
+      setTimeout(() => {
+        inputRef.current?.select();
+      }, 0);
+    }
+  };
+
+  return (
+    <MenuItemStyled py="0.25rem" isSelected={isSelected}>
+      <Box onClick={handleButtonClick} onMouseDown={handleParentMouseDown}>
+        <Group position="apart" px="0.5rem">
+          <Text fw="bold">{`${inputValue} ${name}`}</Text>
+          <NumberInputStyled
+            ref={inputRef}
+            value={inputValue}
+            onChange={(value: number) => setInputValue(value)}
+            onMouseDown={handleChildMouseDownAndUp}
+            onMouseUp={handleChildMouseDownAndUp}
+            onKeyPress={handleEnter}
+            size="xs"
+            w="3.5rem"
+            type="number"
+            required
+          ></NumberInputStyled>
```

Comment:
I _think_ it should be enough to set `onClick={event => event.stopPropagation()}` to avoid conflicts with menu item clicks?

Reply:
It is enough to do that, but those functions have a little more frustration prevention baked into them.

Demo with `onClick={event => event.stopPropagation()}` set (no extra functionality)

https://github.com/metabase/metabase/assets/22608765/f7a55069-672a-4ee9-835a-acc90996cb1c

It does not select all text when a user clicks on it (to make inputting a new number easier). It causes accidental clicks in some scenarios as well: 
1. mouse down on button, but mouse up on input -> still counts as a click
2. mouse down on input, drags across input to select all, but accidentally mouses up on button -> counts as a click

With the extra logic, it will:
1. Select all the text present when a user clicks on it to allow for easier typing.
3. Prevent accidental clicking if a user tries to highlight all the text but accidentally mouses up outside the input on the button
4. Prevents accidental clicking if the user mouses down on the button but mouses up on the input (meaning the user wanted to alter the value not click the button)